chia/data_layer/data_store.py

```diff
@@ -154,12 +186,35 @@ async def _insert_internal_node(self, left_hash: bytes32, right_hash: bytes32, t
             right_hash=right_hash.hex(),
             key=None,
             value=None,
-            tree_id=tree_id.hex(),
         )
 
         return node_hash  # type: ignore[no-any-return]
 
-    async def _insert_terminal_node(self, key: bytes, value: bytes, tree_id: bytes32) -> bytes32:
+    async def _insert_ancestor_table(
+        self,
+        left_hash: bytes32,
+        right_hash: bytes32,
+        tree_id: bytes32,
+        generation: int,
+    ) -> None:
+        node_hash = Program.to((left_hash, right_hash)).get_tree_hash(left_hash, right_hash)
+
```

Comment:
we need to handle the case where the value is already in the table in the case of a failed download that left unreferenced nodes in the db

Reply:
fixed, good catch